subsys/net/lib/openthread/platform/uart.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2020 Tridonic GmbH & Co KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_OPENTHREAD_LOG_LEVEL
+#define LOG_MODULE_NAME net_otPlat_uart
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <kernel.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <drivers/uart.h>
+
+#include <sys/ring_buffer.h>
+#include <sys/atomic.h>
+
+#ifdef CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_ACM
+#include <usb/usb_device.h>
+#endif
+
+#include <openthread-system.h>
+#include <openthread/platform/uart.h>
+
+#include "platform-zephyr.h"
+
+struct openthread_uart {
+	struct ring_buf *tx_ringbuf;
+	struct ring_buf *rx_ringbuf;
+	struct device *dev;
+	atomic_t tx_busy;
+	atomic_t tx_finished;
+};
+
+#define OT_UART_DEFINE(_name, _ringbuf_size) \
+	RING_BUF_DECLARE(_name##_tx_ringbuf, _ringbuf_size); \
+	RING_BUF_DECLARE(_name##_rx_ringbuf, _ringbuf_size); \
+	static struct openthread_uart _name = { \
+		.tx_ringbuf = &_name##_tx_ringbuf, \
+		.rx_ringbuf = &_name##_rx_ringbuf, \
+	};
+
+OT_UART_DEFINE(ot_uart, CONFIG_OPENTHREAD_NCP_UART_RING_BUFFER_SIZE)
+
+#define RX_FIFO_SIZE 128
+
+static void uart_rx_handle(/*const struct openthread_uart *ot_uart*/)
```

Comment:
`static void uart_rx_handle(void)`, let's not leave a dead code.

Reply:
fixed.